### PR TITLE
DOC: Improve anova doc strings

### LINKED ISSUE #3306: ANOVA documentation clarity.
Hi,

I just have a small issue. I'm trying to find the p-value of an ANOVA test. The code worked perfectly and returned a DataFrame with the  final column labeled as "PR(>F)". When I went to double check if this was the same as a p-value, I noticed that the documentation for what is returned by the ANOVA function appears to be missing:

https://github.com/statsmodels/statsmodels/blob/master/statsmodels/stats/anova.py#L293

Cheers!

### PR DESCRIPTION
**Pull request by bashtage**
Provide list of output for ANOVA docstring

closes #3306

### CODE REVIEW

File: statsmodels/stats/anova.py [hunk 1]
```diff
@@ -284,18 +284,25 @@ def anova_lm(*args, **kwargs):
...
     anova : DataFrame
-    A DataFrame containing.
+        A DataFrame with columns:
+
+        - 'sum_sq': Sum of squared residuals
+        - 'df': Difference in sum of squared residuals
+        - Test Name: this column has a name that depends on the test used, for
+          example 'F'.  It contains the value of the test statistic.
+        - Pvalue: The column has a name that depends on the type of the test,
+          for example PR(>F).  Contains the p-value of the test.
```
Comment (josef-pkt) [inline statsmodels/stats/anova.py]: according to line 357 below the columns are
names = ['df_resid', 'ssr', 'df_diff', 'ss_diff', test, pr_test]
?

Comment (josef-pkt) [reply statsmodels/stats/anova.py]: e.g cel 16 in http://www.statsmodels.org/devel/examples/notebooks/generated/interactions_anova.html

**Comment by bashtage** [comment]
```python
import statsmodels.api as sm
from statsmodels.formula.api import ols
moore = sm.datasets.get_rdataset('Moore', 'car', cache=True) # load
data = moore.data
data = data.rename(columns={'partner.status' :
                            'partner_status'}) # make name pythonic
moore_lm = ols('conformity ~ C(fcategory, Sum)*C(partner_status, Sum)',
                data=data).fit()
table = sm.stats.anova_lm(moore_lm, typ=2) # Type 2 Anova DataFrame
print(table)
```

```

                
                                              sum_sq    df          F  \
C(fcategory, Sum)                          11.614700   2.0   0.276958   
C(partner_status, Sum)                    212.213778   1.0  10.120692   
C(fcategory, Sum):C(partner_status, Sum)  175.488928   2.0   4.184623   
Residual                                  817.763961  39.0        NaN   

                                            PR(>F)  
C(fcategory, Sum)                         0.759564  
C(partner_status, Sum)                    0.002874  
C(fcategory, Sum):C(partner_status, Sum)  0.022572  
Residual                                       NaN  
```

File: statsmodels/stats/anova.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/stats/anova.py]: Depends on what is run.  Single anova aren't the same as multiple.

**Comment by bashtage** [comment]
Closing since module need non-trivial rewrite.

**Pull request closed without merging**